src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1564,6 +1582,13 @@ internal void ExecuteAndCatchIfThrows<TArg>(DiagnosticAnalyzer analyzer, Action<
             {
                 ExecuteAndCatchIfThrows_NoLock(analyzer, analyze, argument, info);
             }
+
+            if (_analyzerExecutionTimeMap != null)
+            {
+                var elapsed = timer.Elapsed.Ticks;
+                StrongBox<long> totalTicks = _analyzerExecutionTimeMap.GetOrAdd(analyzer, _ => new StrongBox<long>(0));
+                Interlocked.Add(ref totalTicks.Value, elapsed);
+            }
```

Comment:
📝 Added code was just moved from the previous location.

Reply:
We should likely make `ExecuteAndCatchIfThrows_NoLock` a local function, so if in future there are more callsites added for it, the timing data still accounts for them.

Reply:
new and original code seem odd. should we have this in a finally?  seems weird to start a timer and not complete/report it in some cases (like cancellation)

Reply:
(note: this is a non-blocking concern of mine as the original code had this issue too)

Reply:
> seems weird to start a timer and not complete/report it in some cases (like cancellation)

Results are not observable in cancellation cases.